packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -1442,17 +1447,25 @@ export class DATE extends AbstractDataType<AcceptedDate> {
   /** @hidden */
   static readonly [kDataTypeIdentifier]: string = 'DATE';
   readonly options: DateOptions;
+  protected readonly HAS_TIMEZONE = /(?:Z|[+-]\d{2}(?::\d{2}){0,1}){0,1}$/;
+
+  protected readonly IS_ISO8601 =
+    /^\d{4}(?:-\d{2}){0,2}(?:(?:T| )\d{2}(?::\d{2}){1,2}){0,1}(?:\.\d{1,}){0,1}(?:Z|[+-]\d{2}(?::\d{2}){0,1}){0,1}$/;
+
+  protected readonly IS_PLAIN_DATE =
+    /^\d{4}(?:-\d{2}){0,2}(?:(?:T| )\d{2}(?::\d{2}){1,2}){0,1}(?:\.\d{1,}){0,1}$/;
```

Comment:
Where do these regular expressions come from? Can we add the source as a comment here?

Reply:
I created them based off https://www.w3.org/TR/NOTE-datetime.
I've added a comment now.